tensorflow/core/kernels/cwise_op_select.cc

```diff
@@ -143,21 +143,138 @@ class SelectOp : public OpKernel {
  private:
   TF_DISALLOW_COPY_AND_ASSIGN(SelectOp);
 };
+template <typename Device, typename T>
+class SelectV2Op : public OpKernel {
+ public:
+  explicit SelectV2Op(OpKernelConstruction* context) : OpKernel(context) {}
+
+  void Compute(OpKernelContext* ctx) override {
+    const Tensor* cond;
+    const Tensor* then;
+    const Tensor* else_;
+    OP_REQUIRES_OK(ctx, ctx->input("condition", &cond));
+    OP_REQUIRES_OK(ctx, ctx->input("t", &then));
+    OP_REQUIRES_OK(ctx, ctx->input("e", &else_));
 
-#define REGISTER_SELECT(type)                                      \
-  REGISTER_KERNEL_BUILDER(                                         \
-      Name("Select").Device(DEVICE_CPU).TypeConstraint<type>("T"), \
-      SelectOp<CPUDevice, type>);
+    // The `cond`, `then`, and `else` are broadcastable (bcast.IsValid()),
+    // This matches the behavior of numpy.
+    // TODO (yongtang): Consolidate into n-ary broadcast, instead of multiple
+    // 2-ary broadcast.
+
+    // Combine `then` and `else`.
+    BCast elem_bcast(BCast::FromShape(then->shape()),
```

Comment:
maybe call this then_else_bcast?

Reply:
Thanks @rmlarsen, the name has been changed.